dbt/adapters/glue/impl.py

```diff
@@ -104,22 +104,24 @@ def get_connection(self):
 
     def list_schemas(self, database: str) -> List[str]:
         session, client = self.get_connection()
-        responseGetDatabases = client.get_databases()
-        databaseList = responseGetDatabases['DatabaseList']
+        paginator = client.get_paginator('get_databases')
         schemas = []
-        for databaseDict in databaseList:
-            databaseName = databaseDict['Name']
-            schemas.append(databaseName)
+        for page in paginator.paginate():
+            databaseList = page['DatabaseList']
+            for databaseDict in databaseList:
+                databaseName = databaseDict['Name']
+                schemas.append(databaseName)
         return schemas
 
     def list_relations_without_caching(self, schema_relation: SparkRelation):
         session, client = self.get_connection()
         relations = []
+        tables = []
+        paginator = client.get_paginator('get_tables')
         try:
-            response = client.get_tables(
-                DatabaseName=schema_relation.schema,
-            )
-            for table in response.get("TableList", []):
+            for table in paginator.paginate(DatabaseName=schema_relation.schema):
+                tables.extend(table.get('TableList', []))
+            for table in tables:
```

Comment:
instead of doing another loop here, why don't you do an inner for loop when you get the tabe_page? might be more efficient instead of re-looping a full list of tables again.

Reply:
I see what you're saying, but your version has a bug making and it is less efficient. Namely, we only want to iterate over the table list once, so, consider the case of three pages; your code would:
- append three relations for each table in the first page
- append two relations for each table in the second page

Nonetheless, it would be better if my logic was mirrored to `list_schemas`. I think we want something like this:
```python
            for page in paginator.paginate(DatabaseName=schema_relation.schema):
                for table in page.get('TableList', []):
                    relations.append(self.Relation.create(
                        database=schema_relation.schema,
                        schema=schema_relation.schema,
                        identifier=table.get("Name"),
                        type=self.relation_type_map.get(table.get("TableType")),
                    ))
```

PR has been updated with the above changes.

Reply:
I specifically checked the changed code not all the all function. And yes, I agree, looping the less amount of time possible is the main core idea of my initial comment. What you suggested LGTM.

Reply:
No problem, I understood what you were getting at. Thank you for the review!